src/plugins/auto/auto_executable_network.cpp

```diff
@@ -172,6 +172,8 @@ IE::Parameter AutoExecutableNetwork::GetMetric(const std::string& name) const {
         }
         return decltype(ov::available_devices)::value_type {exeDevices};
     } else if (name == ov::model_name) {
+        if (_autoSchedule->_loadContext[CPU].isEnabled && _autoSchedule->_loadContext[CPU].isAlready)
+            return _autoSchedule->_loadContext[CPU].executableNetwork->GetMetric(name);
```

Comment:
add lock and cases here

Reply:
add lock first and will implement the test case in another PR.